### DISCUSSION TITLE
color output does not work for me

### DISCUSSION THREAD

**Discussion by nepomuk16321** [eyeonus/Trade-Dangerous#163] [General]
The color output does not work for me, neither in the CMD window nor in the PowerShell window.
In the PS window, only the command line is displayed in color.

**Comment by kfsone** [comment]
- What version of TD,
(command: `pip freeze tradedangerous`)
- What version of Windows,
(Settings -> System -> About)
- Are you using Microsoft Windows Terminal or the old console windows?
( Terminal has a tab bar above the console area. https://apps.microsoft.com/detail/9n0dx20hk701 )
- What is the output of running the command in your current window?
```
python -m rich
```
and
```
python -m rich.color
```

**Reply by nepomuk16321** [reply]
_Version of TD:_
appJar==0.94.0
certifi==2024.2.2
charset-normalizer==3.3.2
idna==3.7
ijson==3.2.3
markdown-it-py==3.0.0
mdurl==0.1.2
Pygments==2.17.2
requests==2.31.0
rich==13.7.1
tradedangerous==11.1.6
urllib3==2.2.1

Windows 10 Pro (64bit) 22H2

I use Windows PowerShell (blue background), the successor to the CMD window (black background).

`python -m rich` > Text, text blocks, background and borders are color-coded.

`python -m rich.color` > On the left side color bar 1 to 255, next to it the color name/description (255 = gray93)

I hope the results help.

**Comment by eyeonus** [comment]
This screenshot you posted on the forum has color output:
![image](https://github.com/eyeonus/Trade-Dangerous/assets/1733600/eafbbeb1-67bf-4c02-833c-cd598107c31c)

**Reply by nepomuk16321** [reply]
Yes, you're right, but it's just individual words. There are also words in different colours in the command line.
Perhaps I have misunderstood. In any case, the lists are monochrome. I haven't tried the ‘RUN’ command yet.
There, individual coloured words would be sufficient for a better overview. Ok, I'll test that later today.

**Comment by kfsone** [comment]
For clarification: I recently added the [rich](https://rich.readthedocs.io/en/stable/introduction.html) library to Trade-Dangerous, and made it accessible to TD code without rigorously imposing it on everything (*1).

Doing that gives you various things immediately for free, including some basic syntax highlighting in output. It's also the library many other Python tools rely on for progress bars and spinners etc. 

TD already had some ancient color features ("--color") that I never got round to fleshing out, and I wouldn't expect anyone else to want to either(*2), especially when you have so many different contexts to support(*3). Much better to delegate that responsibility/overhead to rich.


--

(*1 all TD code is supposed to write output via 'tdenv', but not all of it does, and only the code that does is currently taking advantage of rich. the idea is to eliminate the maintenance head of lots of custom terminaly-stuff currently coded into TD. Amongst other things, rich performs platform-compatibility and doing things like not writing color codes to files or between processes)
(*2 it involves hard-coding ansi escape codes which is easy to mess up across different terminals/systems; one terminal's blink is another terminals beep)
(*3 normal command line operation, writing to a file, pipe through another process, running within the simple gui, etc)

**Comment by kfsone** [comment]
@nepomuk16321 Could you post a screenshot showing what you mean? Where you are expecting color that you aren't seeing it, the command line you are using?

**Reply by nepomuk16321** [reply]
I thought that all commands with the ‘--color’ switch also produce coloured output, e.g. the ‘trade’ command.
![image](https://github.com/eyeonus/Trade-Dangerous/assets/20616775/90d2306d-82bb-4c79-9029-29a6c0fc5d4e)
If I enter the command in the command line, the commands and arguments are displayed in colour.
![image](https://github.com/eyeonus/Trade-Dangerous/assets/20616775/c0e83fe4-6e0d-4841-ae40-e30eaa12ce93)
There is then no colour in the output. I thought the output would also be coloured.
![image](https://github.com/eyeonus/Trade-Dangerous/assets/20616775/cfb3e185-aaa5-4380-9628-549191ecfc7a)
And with the ‘RUN’ command (a coloured output would probably make sense here), the output is also only monochrome.
![image](https://github.com/eyeonus/Trade-Dangerous/assets/20616775/c8473f3a-a92f-420c-ae58-37e6cfe122ee)
The colour edition would be nice, but is not necessary. The function is more important!

**Comment by eyeonus** [comment]
The `--color` option was only ever implemented for `trade run`, adding color to the other stuff is an in progress thing currently.

You don't see color output on your `trade run` screenshot because you didn't use the option.

**Reply by nepomuk16321** [reply]
Upps ... You are right. I had tried `--color` with the other commands but not with “**RUN**”, sorry.

**Reply by eyeonus** [reply]
No worries. It's an understandable oversight.
